tests/testthat/test-shinytest2-init.R

```diff
@@ -46,3 +46,58 @@ testthat::test_that("e2e: init creates UI containing specified title, favicon, h
   )
   app$stop()
 })
+
+testthat::test_that("e2e: teal app initializes with Show R Code modal", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = example_module(label = "Example Module")
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check if button exists.
+  button_selector <- sprintf("#%s-rcode-button", app$active_module_ns())
+  testthat::expect_equal(
+    app$get_text(button_selector),
+    "Show R code"
+  )
+
+  app$click(selector = button_selector)
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check header and title content.
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-header > h4"),
+    "Example Code"
+  )
+  testthat::expect_equal(
+    app$get_text(sprintf("#%s-rcode-copy_button1", app$active_module_ns())),
+    "Copy to Clipboard"
+  )
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-body > div > button:nth-child(2)"),
+    "Dismiss"
+  )
+
+  # Check R code output.
+  r_code <- app$get_text(sprintf("#%s-rcode-verbatim_content", app$active_module_ns()))
+
+  greplf <- function(text, content = r_code) {
+    grepl(text, content, fixed = TRUE)
+  }
+
+  testthat::expect_true(greplf("# Add any code to install/load your NEST environment here"))
+  testthat::expect_true(greplf("library(teal.code)"))
+  testthat::expect_true(greplf("stopifnot(rlang::hash("))
```

Comment:
What do you think about using `expect_match` instead?

```suggestion
  testthat::expect_match(r_code, "# Add any code to install/load your NEST environment here", fixed = TRUE)
  testthat::expect_match(r_code, "library(teal.code)", fixed = TRUE)
  testthat::expect_match(r_code, "stopifnot(rlang::hash(", fixed = TRUE)
```